include/pybind11/async.h

```diff
@@ -0,0 +1,205 @@
+#pragma once
+
+#include <memory>
+#include <future>
+#include <chrono>
+
+#include "pybind11/pybind11.h"
+
+
+namespace py = pybind11;
+
+
+PYBIND11_NAMESPACE_BEGIN(PYBIND11_NAMESPACE)
+PYBIND11_NAMESPACE_BEGIN(async)
+
+
+class StopIteration : public py::stop_iteration {
+    public:
+        StopIteration(py::object result) : stop_iteration("--"), result(result) {};
+
+        void set_error() const override {
+            PyErr_SetObject(PyExc_StopIteration, this->result.ptr());
+        }
+    private:
+        py::object result;
+};
+
+
+class Awaitable {
+    public:
+        Awaitable() {
+            this->future = std::future<py::object>();
+        };
+
+        Awaitable(std::future<py::object>& _future) {
```

Comment:
Shouldn't this just be a const ref copy constructor which would negate the need for std::move?